src/solc_0.7/catalyst/GemsCatalystsRegistry.sol

```diff
@@ -54,7 +54,7 @@ contract GemsCatalystsRegistry is WithSuperOperators, WithMetaTransaction, IGems
         uint256 amount
     ) external override {
         for (uint256 i = 0; i < gemIds.length; i++) {
-            // TODO optimize it by checking adjacent gemIds
+            // @todo optimize it by checking adjacent gemIds
```

Comment:
Looking at this now.

Reply:
I guess we could parse the array of gemIds, and call `batchBurnGems` with a batch of each type of gem present.
It's not immediately clear to me whether this would achieve much in the way of gas savings though...

Reply:
Actually, I see that it would save us gas.

Reply:
@wighawag Looking at this again today, I'm not sure there's too much to be gained here?
We already have a `batchBurnCatalysts` function, the main difference being that it allows the burning of different amounts for each catalystId, whereas for `burnDifferentCatalysts` one must specify a fixed amount to burn for all catalystIds. The easiest optimization here may be to inspect the array of ids passed, and if the same id is found in more than one index, revert with something like "ID_ARRAY_FORMATTING_ERROR" . ie: this `[3, 1, 2, 1]` would fail, as the common catalyst id `1` is specified in two places. FOr this set of cataystIds, the `batchBurnCatalysts` function is a better fit.

Reply:
yes, we can I think leave as is. I was confused with AssetMinter use of `_batchBurnGems`
We already optimize it vy asking the user to pass GemQuantities